xen/arch/x86/include/asm/intel_txt.h

```diff
@@ -388,11 +387,6 @@ static inline void txt_verify_pmr_ranges(struct txt_os_mle_data *os_mle,
     if ( !is_in_pmr(os_sinit, load_base_addr, xen_size, check_high_pmr) )
         txt_reset(SLAUNCH_ERROR_LO_PMR_MLE);
 
-    /* Check if all of Xen after relocation is covered by PMR. */
```

Comment:
For legacy, relocation (if required) happens after this function is called, and Xen memory is measured before. This means that both ranges must be protected.

For UEFI, IIUC no relocation takes place after DL Event, so one check is sufficient there. The easiest way to achieve this would be to set `load_base_addr` and `tgt_base_addr` to the same value.

Reply:
https://github.com/TrenchBoot/xen/compare/97a6affd50d6126c0b77e1e5ce88c0bb65823b14..1a33e907a021e6192b6fce3811fd977e44f3cf8e